### PR TITLE
Implement edit-url to craft a url with an editor.

### LINKED ISSUE #1261: Open the current url in an editor
If I want to modify the current url, it might be nice to have a command that opens the current url in an editor and navigates to that url when the editor is closed. 

Think of something like visual mode when using vi mode for bash -- you press `v` and the current command line is opened in vi. When you close the editor, the command is run.

Right now, it seems like I have to press `O` (to fill the command line with the current url), backspace a lot, then retype most of the url.

### PR DESCRIPTION
**Pull request by rcorre**
The edit-url command opens a url (by default, the current url) in the
user's external editor and navigates to the result when the editor is
closed. This makes it easy to tweak the current url to navigate within
a site.

`edit-url` accepts the same flags as `open` (e.g. -t will open in a new
tab.

One may provide a url as an argument to create a shortcut to
pre-populate part of a url and allow filling in the rest.

There is no default keybinding.

Resolves #1261.

### CODE REVIEW

**Comment by The-Compiler** [comment]
Can you try without setting `self._editor`? To be honest I'm not even sure anymore why I did set it with `:open-editor` - the only reason I can imagine is it being collected by Python's garbage collector, but that shouldn't happen with `self._tabbed_browser` passed as parent.

File: qutebrowser/browser/commands.py [hunk 1]
```diff
@@ -1847,3 +1847,27 @@ def debug_clear_ssl_errors(self):
...
+            window: Open in a new window.
+            count: The tab index to open the URL in, or None.
+        """
+        self._editor = editor.ExternalEditor(
+            self._win_id, self._tabbed_browser)
+
+        # Passthrough for openurl args (e.g. -t, -b, -w)
+        self._editor.editing_finished.connect(
+            functools.partial(self.openurl,
+            bg = bg, tab = tab, window = window, count = count))
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: Please remove the spaces around `=` here, as [per pep8](https://www.python.org/dev/peps/pep-0008/#other-recommendations) (as they are keyword arguments, not assignments)

**Comment by rcorre** [comment]
`open-editor` seems fine without setting `self._editor` as well -- I wrote this message with it :)
I can remove `self._editor` as part of this PR if you like.

**Comment by The-Compiler** [comment]
Thanks! I'll remove them myself, first want to check the git history and why I added them :wink:

**Comment by The-Compiler** [comment]
Merged! I [wrote some tests for this](https://github.com/The-Compiler/qutebrowser/commit/86dec02fe806dd5f25147c667ef24dbf86f00c7f) and discovered/fixed some issues which weren't immediately obvious:

- Opening an invalid URL crashed because `cmdexc.CommandError` was raised in `openurl`: c4878bb7ed40f9e2682fe467d9efba9184db2392
- Using `:edit-url -t -b` crashed because it was only validated later in `openurl` too: 776c4c4400b2b4c15c3104403151bb44d16d53ce
- `count` wasn't marked as count and treated like a normal argument: fc921377063b198697c6e396f662dd156007e15c

I also [removed `self._editor` now](https://github.com/The-Compiler/qutebrowser/commit/9db697452da3c9db619a49c231a3f0cfa298ac59).

**Comment by rcorre** [comment]
Thank you! I was looking at the tests but wasn't sure where it would fit.

**Comment by The-Compiler** [comment]
Yeah, for some things there simply isn't the necessary test "infrastructure" there yet (like the `set_up_editor_replacement` with the fake editor) - then I usually accept PRs without tests.

Fortunately, it's getting better and better - ideally I want to arrive at a point where it's trivial to add a test for any new feature :wink:

**Pull request merged**